### PR TITLE
Bngp 188 legal agreement pre upload questions

### PR DESCRIPTION
**Pull request by shaundefra**
Add legal agreement pre-upload question tests and refactor

### CODE REVIEW

File: src/features/page_objects/legal_agreement/add-legal-agreement-parties [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    return $(".govuk-input");
+  }
+    
+  get legalPartyRole(){
+    //Todo: fix when html on page is fixed and all ids are unique
+    return $(".govuk-radios__input");
+  }
+
+  get legalContinueButton() {
+    return $("#continue");
```
Comment (timstone) [inline src/features/page_objects/legal_agreement/add-legal-agreement-parties:19]: Do you have different selectors for continue/submit form buttons on each page or could you just have one in the base page?

File: src/features/step_definitions/common.steps.js [hunk 2]
```diff
@@ -64,6 +74,67 @@ When("I continue without an action", async () => {
...
+When("I enter a valid startdate of {string} for the {string} page", async (date, page) => {
+
+  var arr = date.split('/');
+
+  switch (page) {
+    case "habitat-works-start-date": {
+          await habitatWorksStartDatePage.Day.addValue(arr[0]);
+          await habitatWorksStartDatePage.Month.addValue(arr[1]);
+          await habitatWorksStartDatePage.Year.addValue(arr[2]);
+          await (await habitatWorksStartDatePage.continueButton).click();
```
Comment (timstone) [inline src/features/step_definitions/common.steps.js:121]: if you have all these selectors in the base page class then you won't have to duplicate this code and you don't have to have the funky `for the X page` part in your step

File: src/features/page_objects/legal_agreement/add-legal-agreement-parties (hunk 1 above)
Comment (shaundefra) [reply src/features/page_objects/legal_agreement/add-legal-agreement-parties:19]: Ive had to ask the dev to add a generic id for it as currently its been added seperately

File: src/features/step_definitions/common.steps.js (hunk 2 above)
Comment (shaundefra) [reply src/features/step_definitions/common.steps.js]: Have updated the selectors that I can control instead

**Review by timstone — APPROVED** [review]
startdate typo

**Pull request merged**